### DISCUSSION TITLE
Amazon SES SMTP settings for Mastodon

### DISCUSSION THREAD

**Discussion by olivievranska** [mastodon/mastodon#21923] [Show and tell]
Hello,

I spent couple of hours by finding how to make Amazon SES working for e-mails sent from Mastodon.

I hope these information will help anyone else.

**1. Register/logon to Amazon AWS, add Amazon SES service.**
Create identity, set DNS records for DKIM and Custom MAIL FROM domain, wait for validation.
Create SMTP credentials, write down generated username and password.
Request productive version of Amazon SES.
Send test e-mail - go to Verified identities, open it and click on Send test email. Try both formats, fill e-mail address defined in SMTP_FROM_ADDRESS, select Scenario Successful delivery, put existing e-mail you can access to Cc: (part Additional configuration) to ensure successfully delivery. Check if test e-mail will come.

**Write down:**
SMTP endpoint value (Amazon SES - SMTP settings)
SMTP credentials - generated username, generated password created above
MAIL FROM domain value (Verified identities - Custom MAIL FROM domain)
username to be used as sender (eg. notifications, mastodon, etc.)

**2. Logon to Mastodon instance via SSH, switch to user mastodon, go to /home/mastodon/live and open configuration file .env.production.**
Fill SMTP settings:

SMTP_SERVER=_SMTP endpoint_
SMTP_PORT=465
SMTP_LOGIN= _generated username_
SMTP_PASSWORD= _generated password_
SMTP_ENABLE_STARTTLS_AUTO=false
SMTP_TLS=true
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_FROM_ADDRESS='Mastodon @instance-domain <_sender-username_@_MAIL FROM domain_>'

Save, exit.
Restart mastodon-* services.

**Comment by pennydoesdev** [comment]
I am afraid this does not work for me

**Comment by blazer82** [comment]
I also added

```
SMTP_ENABLE_STARTTLS=never
```

not sure if necessary but it works 😊
